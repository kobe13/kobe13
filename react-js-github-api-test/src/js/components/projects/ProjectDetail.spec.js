import * as React from 'react';
import renderer from 'react-test-renderer';
import ProjectView from './ProjectDetail';

test('Project detail with a homepage renders correctly', () => {
  const projectDetail = [
    {
      name: 'project name',
      description: 'Description here',
      language: 'React JS',
      homepage: 'https://test-url.com',
      key: 1,
    },
  ];
  const component = renderer.create(<ProjectView projects={projectDetail} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Project detail without a homepage renders correctly', () => {
  const projectDetail = [
    {
      name: 'project name',
      description: 'Description here',
      language: 'React JS',
      key: 1,
    },
  ];
  const component = renderer.create(<ProjectView projects={projectDetail} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
