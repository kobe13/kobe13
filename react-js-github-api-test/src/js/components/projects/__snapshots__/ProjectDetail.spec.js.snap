// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project detail with a homepage renders correctly 1`] = `
<div>
  <div>
    <h2>
      <a
        href="https://test-url.com"
        title="Project homepage"
      >
        project name
      </a>
    </h2>
    <p>
      Description here
    </p>
    <p>
      <i>
        <b>
          Language
        </b>
        : 
        React JS
      </i>
    </p>
  </div>
</div>
`;

exports[`Project detail without a homepage renders correctly 1`] = `
<div>
  <div>
    <h2>
      project name
    </h2>
    <p>
      Description here
    </p>
    <p>
      <i>
        <b>
          Language
        </b>
        : 
        React JS
      </i>
    </p>
  </div>
</div>
`;
